CODE_GUIDELINES.md

```diff
@@ -12,12 +12,14 @@ Instead, write components with generic objects in mind, like people, media, acti
 
 Component js, style, test, and story should be siblings. If they're in their own directory, provide an `index.js` file.
 
-```
+```sh
+$ tree src/components/Modal
 Modal/
-|-- index.js   # Core component file, imports style.css
-|-- style.css  # Styles, as CSS modules
-|-- test.js    # Used with an npm command
-`-- story.js  # Used by storybook, separate npm command
+|-- index.ts          # Re-exports from Modal.tsx
+|-- Modal.tsx         # Core component file, imports Modal.css
```

Comment:
Bikeshedding:
I know I read a couple of places that newer recs were to use `.js` and not `.jsx`, even with JSX in the files.
The argument boiled down to "everything is transpiled anyway, and not having to rename files is better ergonomics".

Would that make sense with TS files too (skip the trailing "x"?)
I don't mind much either way, but I have a slight preference for using the same extension for all.

Reply:
I have not looked into this much, but when I leave off the `s` for TS components, TS definitely complains loudly. With the `x`, it happily parses it.

Reply:
@pixelbandito Ah, here's an explanation for this: https://www.typescriptlang.org/docs/handbook/jsx.html#basic-usage. For explicit TS, we need to use the `.tsx` extension to enable JSX for TypeScript components